src/care/LoopFuser.h

```diff
@@ -339,6 +339,266 @@ class SerializableDeviceLambda {
+      }
+      m_is_scan = scan;
+   }
+
+   ///////////////////////////////////////////////////////////////////////////
+   /// @brief set counts_to_offsets_scan mode
+   ///////////////////////////////////////////////////////////////////////////
+   virtual void setCountsToOffsetsScan(bool scan) {
+      for (auto & action_priority : m_fused_actions) {
+         action_priority.first->setCountsToOffsetsScan(scan);
+      }
+      m_is_counts_to_offsets_scan = scan;
+   }
+
+
+   void flushActions(bool async) {
+#ifdef FUSER_VERBOSE
+      printf("Observer flushActions()\n");
+#endif
+      for (auto & priority_action : m_fused_action_order) {
+         FusedActions * const & actions = priority_action.second;
+#ifdef FUSER_VERBOSE
+         printf("Observer::flushActions::action count %i\n",actions->actionCount());
+#endif
+         if (actions -> actionCount() > 0) {
+#ifdef FUSER_VERBOSE
+            printf("Observer::flushActions priority:%g ptr:%p\n",priority_action.first, actions);
+#endif
+            actions->flushActions(async);
+         }
+      }
+      for (auto & array: m_to_be_freed) {
+         array.free();
+      }
+      m_to_be_freed.clear();
+   }
+
+   void registerFusedActions(FusedActions * actions, double priority) {
+      auto this_iter = m_fused_actions.find(actions);
+      if (this_iter == m_fused_actions.end()) {
+         if (m_recording) {
+            actions->startRecording();
+         } else {
+            actions->stopRecording();
+         }
+         if (m_last_insert_priority >= priority) {
+            printf("CARE: WARNING fused action registered out of priority order\n");
+         }
+         m_fused_action_order[priority] = actions;
+         m_fused_actions[actions] = priority;
+      }
+      m_last_insert_priority = priority;
+
+   }
+
+   void reset_phases() {
+      m_last_insert_priority = -FLT_MAX;
+   }
+
+
+   int actionCount() {
+      int count = 0;
+      for (auto actions_priority : m_fused_actions) {
+         count += actions_priority.first->actionCount();
+      }
+      return count;
+   }
+
+
+   void reset(bool /*async*/) {
+      m_fused_actions.clear();
+      m_fused_action_order.clear();
+      m_to_be_freed.clear();
+      m_recording = false;
+
+   }
+
+
+   ///////////////////////////////////////////////////////////////////////////
+   /// @author Peter Robinson
+   /// @brief registers an array to be released after a flushActions()
+   /// @param[in] array : the array to be freed after a flushActions
+   ///////////////////////////////////////////////////////////////////////////
+   template <typename T>
+   void registerFree(care::host_device_ptr<T> & array) {
+      m_to_be_freed.push_back(reinterpret_cast<care::host_device_ptr<char> &>(array));
+   }
+
+   protected:
+      std::unordered_map<FusedActions *, double> m_fused_actions;
+      std::map<double, FusedActions *> m_fused_action_order;
+      double m_last_insert_priority = -1.0;
```

Comment:
The default value is set here and in the only constructor.  Should we get rid of one of them?

Reply:
done